code/modules/reagents/chemistry/holder/holder.dm

```diff
@@ -532,9 +532,14 @@
 		remove_reagent(reagent.type, transfer_amount)
 		transfer_log[reagent.type] = list(REAGENT_TRANSFER_AMOUNT = transfer_amount, REAGENT_PURITY = reagent.purity)
 
-	if(transferred_by && target_atom)
-		target_atom.add_hiddenprint(transferred_by) //log prints so admins can figure out who touched it last.
-		log_combat(transferred_by, target_atom, "transferred reagents ([get_external_reagent_log_string(transfer_log)]) from [my_atom] to")
+	if(transferred_by)
+		if(isorgan(target_atom))
+			var/obj/item/organ/target_organ = target_atom
+			target_organ.owner?.add_hiddenprint(transferred_by) //log prints so admins can figure out who touched it last.
+			log_combat(transferred_by, target_organ.owner ? target_organ.owner : target_organ, "transferred reagents to", my_atom, "which had [get_external_reagent_log_string(transfer_log)]")
+		else
+			target_atom?.add_hiddenprint(transferred_by) //log prints so admins can figure out who touched it last.
+			log_combat(transferred_by, target_atom, "transferred reagents to", my_atom, "which had [get_external_reagent_log_string(transfer_log)]")
```

Comment:
rather then duplicating the code like this, store as a var the thing we're transferring into, and set it in the isorgan case. much cleaner, easier to expand on in future.

Reply:
yeah, that's nicer